### PR TITLE
Apache License Version 2.0

### PR DESCRIPTION
**Pull request by eed3si9n**
Fixes https://github.com/scala/bug/issues/9026

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -76,16 +76,30 @@ lazy val publishSettings : Seq[Setting[_]] = Seq(
...
+ThisBuild / startYear      := Some(2002)
+ThisBuild / licenses       += (("Apache-2.0", url("https://github.com/scala/scala/blob/2.13.x/LICENSE")))
+def epflCopyright          = "(c) 2002-2018 EPFL"
+def lightbendCopyright     = "(c) 2011-2018 Lightbend, Inc."
+ThisBuild / headerLicense  := Some(HeaderLicense.Custom(
+  s"""                     __
+     |     ________ ___   / /  ___     Scala
+     |    / __/ __// _ | / /  / _ |    $epflCopyright
+     |  __\\ \\/ /__/ __ |/ /__/ __ |    $lightbendCopyright
+     | /____/\\___/_/ |_/____/_/ | |    http://scala-lang.org/
```
Comment (sjrd) [inline build.sbt]: https ?

File: build.sbt [hunk 2]
```diff
@@ -76,16 +76,30 @@ lazy val publishSettings : Seq[Setting[_]] = Seq(
 // should not be set directly. It is the same as the Maven version and derived automatically from `baseVersion` and
 // `baseVersionSuffix`.
 globalVersionSettings
-baseVersion in Global := "2.13.0"
-baseVersionSuffix in Global := "SNAPSHOT"
+Global / baseVersion       := "2.13.0"
+Global / baseVersionSuffix := "SNAPSHOT"
+ThisBuild / organization   := "org.scala-lang"
+ThisBuild / homepage       := Some(url("http://www.scala-lang.org"))
```
Comment (sjrd) [inline build.sbt]: https ?

**Comment by SethTisue** [comment]
fixes https://github.com/scala/bug/issues/9026

**Comment by SethTisue** [comment]
if we include the copyright year in every source file, then we'll have to update it every year, and then the history of every source file will become cluttered with those annual commits. would omitting the per-source-file years, and just putting the years in a small number of centralized places, be okay...?

**Comment by eed3si9n** [comment]
@SethTisue As per Apache is concerned that's fine, so let's do that.

**Review by jvican — APPROVED** [review]
Yuhu! Really excited to see this happening.

File: test/files/pos/spec-Function1.scala [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+ *      ________ ___   / /  ___     Scala
+ *     / __/ __// _ | / /  / _ |    (c) EPFL
+ *   __\ \/ /__/ __ |/ /__/ __ |    (c) Lightbend, Inc.
+ *  /____/\___/_/ |_/____/_/ | |    https://www.scala-lang.org
+ *                           |/     Licensed under Apache License 2.0
+ *
+ *  See the NOTICE file distributed with this work for additional information
+ *  regarding copyright ownership.
+ */
+
```
Comment (dwijnand) [inline test/files/pos/spec-Function1.scala]: this doesn't seem necessary

File: test/instrumented/boxes.patch [hunk 4]
```diff
@@ -21,46 +20,46 @@
...
- 
+
      public static java.lang.Byte boxToByte(byte b) {
 +        byteBoxCount += 1;
          return java.lang.Byte.valueOf(b);
      }
- 
+
      public static java.lang.Short boxToShort(short s) {
 +        shortBoxCount += 1;
```
Comment (dwijnand) [inline test/instrumented/boxes.patch:41]: what's happening here?

File: build.sbt [hunk 5]
```diff
@@ -175,6 +187,7 @@ lazy val commonSettings = instanceSettings ++ clearSourceAndResourceDirectories
         </developer>
       </developers>
   },
+  headerLicense := (ThisBuild / headerLicense).value,
```
Comment (dwijnand) [inline build.sbt:190]: why is this necessary? (just, out of curiosity)

File: build.sbt [hunk 6]
```diff
@@ -76,16 +76,31 @@ lazy val publishSettings : Seq[Setting[_]] = Seq(
...
 // `baseVersionSuffix`.
 globalVersionSettings
-baseVersion in Global := "2.13.0"
-baseVersionSuffix in Global := "SNAPSHOT"
+Global / baseVersion       := "2.13.0"
+Global / baseVersionSuffix := "SNAPSHOT"
+ThisBuild / organization   := "org.scala-lang"
+ThisBuild / homepage       := Some(url("https://www.scala-lang.org"))
+ThisBuild / startYear      := Some(2002)
+ThisBuild / licenses       += (("Apache-2.0", url("https://github.com/scala/scala/blob/2.13.x/LICENSE")))
```
Comment (dwijnand) [inline build.sbt]: Is it worth maintaining this URL every major version, or using, perhaps, https://www.apache.org/licenses/LICENSE-2.0 ?

File: test/instrumented/boxes.patch (hunk 4 above)
Comment (dwijnand) [reply test/instrumented/boxes.patch:41]: Oh, nvm, GitHub's inter-diff syntax highlighting confused me.

File: build.sbt (hunk 6 above)
Comment (adriaanm) [reply build.sbt]: +1 on using the apache url.

**Comment by adriaanm** [comment]
See also https://github.com/scala/scala-lang/pull/958

File: build.sbt [hunk 7]
```diff
@@ -76,16 +76,31 @@ lazy val publishSettings : Seq[Setting[_]] = Seq(
...
+ThisBuild / headerLicense  := Some(HeaderLicense.Custom(
+  s"""                     __
+     |     ________ ___   / /  ___     Scala
+     |    / __/ __// _ | / /  / _ |    (c) EPFL
+     |  __\\ \\/ /__/ __ |/ /__/ __ |    (c) Lightbend, Inc.
+     | /____/\\___/_/ |_/____/_/ | |    ${(ThisBuild/homepage).value.get}
+     |                          |/     Licensed under Apache License 2.0
```
Comment (lrytz) [inline build.sbt]: I wouldn't mind having this fine piece of art printed, framed, put on a shelf and move on to a future of lighter headers in scala/scala source files 😊

Comment (adriaanm) [reply build.sbt]: It sure is pretty, but I'm also in favor of a more minimal approach.

File: test/files/pos/spec-Function1.scala (hunk 3 above)
Comment (eed3si9n) [reply test/files/pos/spec-Function1.scala]: We used to have copyright 2012-20XX. This is a replacement for that using the wording used by Apache foundation. https://www.apache.org/legal/src-headers.html

File: build.sbt (hunk 5 above)
Comment (eed3si9n) [reply build.sbt:190]: Because I want to define it at ThisBuild, but the plugin doesn't.

File: test/files/pos/spec-Function1.scala (hunk 3 above)
Comment (dwijnand) [reply test/files/pos/spec-Function1.scala]: But this is a test file, which we don't distribute.

File: build.sbt (hunk 5 above)
Comment (dwijnand) [reply build.sbt:190]: :)

File: test/files/pos/spec-Function1.scala (hunk 3 above)
Comment (eed3si9n) [reply test/files/pos/spec-Function1.scala]: Oh you mean the fact that the header is applied to `files/pos`? Some files are I think tested against source files in main source to test code generation etc, so I just manually applied headers.

File: build.sbt (hunk 7 above)
Comment (SethTisue) [reply build.sbt]: haha I'm so used to it I didn't even occur to me to suggest a change

agree, it isn't worth the vertical real estate, and now's our chance.

Comment (eed3si9n) [reply build.sbt]: It only looks ugly here because I needed to escape backslash to `\\` even though I am using tripplequote `"""`. Is that an expected behavior with `s"""`?

Comment (eed3si9n) [reply build.sbt]: May I interest you in Bloody theme during the month of October?

```
  ██████  ▄████▄   ▄▄▄       ██▓    ▄▄▄      
▒██    ▒ ▒██▀ ▀█  ▒████▄    ▓██▒   ▒████▄    
░ ▓██▄   ▒▓█    ▄ ▒██  ▀█▄  ▒██░   ▒██  ▀█▄  
  ▒   ██▒▒▓▓▄ ▄██▒░██▄▄▄▄██ ▒██░   ░██▄▄▄▄██ 
▒██████▒▒▒ ▓███▀ ░ ▓█   ▓██▒░██████▒▓█   ▓██▒
▒ ▒▓▒ ▒ ░░ ░▒ ▒  ░ ▒▒   ▓▒█░░ ▒░▓  ░▒▒   ▓▒█░
░ ░▒  ░ ░  ░  ▒     ▒   ▒▒ ░░ ░ ▒  ░ ▒   ▒▒ ░
░  ░  ░  ░          ░   ▒     ░ ░    ░   ▒   
      ░  ░ ░            ░  ░    ░  ░     ░  ░
         ░                                   
```

File: test/files/pos/spec-Function1.scala (hunk 3 above)
Comment (dwijnand) [reply test/files/pos/spec-Function1.scala]: Ah, I see.

File: build.sbt (hunk 7 above)
Comment (adriaanm) [reply build.sbt]: Careful not to trigger font bikeshedding 😱

File: build.sbt (hunk 6 above)
Comment (adriaanm) [reply build.sbt]: We should also update our pom generation, and link to apache.org instead of scala-lang.org

File: build.sbt (hunk 7 above)
Comment (eed3si9n) [reply build.sbt]: I took the liberty to replace "NSC -- new Scala compiler" with just "Scala". Are you folks cool with that?

Comment (adriaanm) [reply build.sbt]: Yeah, better not to be reminded it's actually the new old scala compiler :-)

File: project/GenerateAnyVals.scala [hunk 8]
```diff
@@ -273,13 +273,16 @@ import scala.language.implicitConversions"""
 }
 
 trait GenerateAnyValTemplates {
-  def headerTemplate = """/*                     __                                               *\
-**     ________ ___   / /  ___     Scala API                            **
-**    / __/ __// _ | / /  / _ |    (c) 2002-2013, LAMP/EPFL             **
-**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/               **
-** /____/\___/_/ |_/____/_/ | |                                         **
-**                          |/                                          **
-\*                                                                      */
```
Comment (eed3si9n) [inline project/GenerateAnyVals.scala:282]: RIP ASCII art.

Comment (SethTisue) [reply project/GenerateAnyVals.scala:282]: at the risk of bikeshedding (but, we only want to do this once):

the main reason I was interested in seeing the ASCII art die was to save vertical screen real estate, it wasn't actually about the ASCII art itself.

wdyt about condensing

```
/*
 * Scala
 * (c) EPFL
 * (c) Lightbend, Inc.
 * https://www.scala-lang.org
 * Licensed under Apache License 2.0
 *
 * See the NOTICE file distributed with this work for additional information
 * regarding copyright ownership.
 */
```

to

```
/* Scala. (c) EPFL, (c) Lightbend, Inc. Apache License 2.0.
 * See NOTICE file for full license and copyright info. */
```

Comment (dwijnand) [reply project/GenerateAnyVals.scala:282]: Btw, the "standard license header" is quite long: https://spdx.org/licenses/Apache-2.0.html#licenseHeader

Comment (eed3si9n) [reply project/GenerateAnyVals.scala:282]: I am personally meh about saving some vertical lines on top of the file, but I don't work with this code base every day. `/*` at the top and bottom is generated by sbt-header, so it's out my control. If you could tally up the Scala team's opinion on exact header on top, I can make another change. There's adjustment of patch file I have to do etc, so it's not super easy.

Comment (NthPortal) [reply project/GenerateAnyVals.scala:282]: Personally, I don't mind the vertical lines at the top of the file. It's not that long, and honestly feels... like it makes the copyright/license notice less ugly and annoying? idk. that's just how I personally feel.

Also I feel like the ASCII art makes it easy to tell at a glace that it's the copyright header, whereas with the suggested 2 line version, it's less obvious and (without reading) might be a comment about the internal structure of the file or something.

Comment (adriaanm) [reply project/GenerateAnyVals.scala:282]: Let's go with the following semi-compromise (the usual outcome of bikeshedding...):

```
Scala (${(ThisBuild/homepage).value.get})

Copyright EPFL and Lightbend, Inc.

Licensed under the Apache License, Version 2.0
(http://www.apache.org/licenses/LICENSE-2.0).

See the NOTICE file distributed with this work for 
additional information regarding copyright ownership.
```

Comment (adriaanm) [reply project/GenerateAnyVals.scala:282]: The main thing I care about: spell out "Copyright", leave off the year so we don't need to bother updating it. If this ever goes to court, there's plenty of evidence on the blockchain^W git.

Comment (eed3si9n) [reply project/GenerateAnyVals.scala:282]: Done https://github.com/scala/scala/blob/2d9e6acce8c4246e6cba5a22ff9d965ec3bd117c/src/library/scala/Array.scala

Comment (lrytz) [reply project/GenerateAnyVals.scala:282]: Thank you! Let's merge this before we run into merge conflicts @adriaanm

File: doc/LICENSE.md [hunk 9]
```diff
@@ -1,4 +1,4 @@
-Scala is licensed under the [BSD 3-Clause License](http://opensource.org/licenses/BSD-3-Clause).
+Scala is licensed under the [Apache License Version 2.0](https://www.apache.org/licenses/LICENSE-2.0).
 
 ## Scala License
```
Comment (dwijnand) [inline doc/LICENSE.md:3]: Should this be dropped, or reworded?  It's not longer the "Scala License", based on BSD 3-Clause, it's now the Apache Licence, Version 2.0, as is.

**Review by dwijnand — APPROVED** [review]
This is great, and I echo Lukas' thought: let's merge this before it conflicts.

File: project/GenerateAnyVals.scala (hunk 8 above)
Comment (eed3si9n) [reply project/GenerateAnyVals.scala:282]: To counter the encroaching Swiss minimalist contingency I am sending a PR to add ASCII art to sbt/sbt's startup - https://github.com/sbt/sbt/pull/4402/commits/0d788162aef45eb6efea7713208c508074823e4f

**Comment by adriaanm** [comment]
🍰 thanks for getting this done, @eed3si9n!

File: project/GenerateAnyVals.scala (hunk 8 above)
Comment (som-snytt) [reply project/GenerateAnyVals.scala:282]: What a shock to update! It's still broken on 2.13, but now I recognize the historical importance of:
```
$ scala -Dscala.banner=welcome

     ________ ___   / /  ___  
    / __/ __// _ | / /  / _ | 
  __\ \/ /__/ __ |/ /__/ __ | 
 /____/\___/_/ |_/____/_/ | | 
                          |/  version 2.12.6

scala> 
```

Comment (som-snytt) [reply project/GenerateAnyVals.scala:282]: Whew...
```
$ scalam -Dscala.repl.welcome=banner

     ________ ___   / /  ___
    / __/ __// _ | / /  / _ |
  __\ \/ /__/ __ |/ /__/ __ |
 /____/\___/_/ |_/____/_/ | |
                          |/  version 2.13.0-M5

scala> 
```
That should go in the release notes...

**Comment by som-snytt** [comment]
That was quite a-patch-y.
